helm-charts/splunk-otel-collector/templates/service.yaml

```
@@ -20,6 +20,9 @@ metadata:
 {{- end }}
 spec:
   type: {{ .Values.service.type }}
+  {{- if .Values.service.trafficDistribution }}
```

```suggestion
  {{- if and .Values.service.trafficDistribution (semverCompare ">=1.33.0-0" .Capabilities.KubeVersion.Version) }}
```
We could additionally verify in template but the clarification in comment is also good with me.

---

I think it's fine to fail rather then suppressing the error